agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalStorageTelemetryPipelineListener.java

```diff
@@ -48,9 +49,14 @@ public LocalStorageTelemetryPipelineListener(
     LocalFileCache localFileCache = new LocalFileCache(telemetryFolder);
     LocalFileLoader loader =
         new LocalFileLoader(localFileCache, telemetryFolder, stats, suppressWarnings);
-    localFileWriter = new LocalFileWriter(localFileCache, telemetryFolder, stats, suppressWarnings);
+    localFileWriter =
+        new LocalFileWriter(
+            diskPersistenceMaxSizeMb, localFileCache, telemetryFolder, stats, suppressWarnings);
 
-    localFileSender = new LocalFileSender(loader, pipeline, suppressWarnings);
+    // send persisted telemetries from local disk every 30 seconds by default.
+    // if diskPersistenceMaxSizeMb is greater than 50, it will get changed to 10 seconds.
+    long intervalSeconds = diskPersistenceMaxSizeMb > 50 ? 10 : 30;
```

Comment:
estimating how long this will take to send a large number of batches from disk:

  600 characters per telemetry
* 512 telemetry per batch
= ~300kb per batch

1gb of telemetry will be ~3000 batches

so sending 1 batch per 10 seconds, this will take 8 hours

which is not bad

(we can always revisit this detail when spec'ing it out with other langs)

Reply:
i can a todo to revisit this interval.

Reply:
@trask We allow (in .NET) users to configure how many parallel sends can occur. (Default is 10). They can increase it to a large number, increasing the throughput heavily, subject to physical limitations of the network itself. 

While scenarios like load testing, by increasing the senders, user could send HUGE amounts of data.